components/Table/Table.js

```diff
@@ -95,12 +118,186 @@ export class Table {
+    filterValue.classList.add('filter-value');
+    filterValue.placeholder = 'Filter value';
+
+    const btnClean = document.createElement('div');
+    btnClean.classList.add('btn-clean');
+    btnClean.textContent = 'x';
+
+    const btnExit = document.createElement('div');
+    btnExit.classList.add('btn-exit');
+    btnExit.textContent = 'x';
+
+    btnExit.addEventListener('click', () => {
+      filterMenuContainer.remove();
+    });
+
+    btnClean.addEventListener('click', () => {
+      filterValue.value = '';
+      if (this.callbacksMap.sortCallback) {
+        this.callbacksMap.sortCallback();
+      }
+    });
+
+    const filterFunction = (e) => {
+      let filter = e.target.value;
+      if (this.callbacksMap.filterCallback) {
+        this.callbacksMap.filterCallback(filter);
+      }
+    };
+
+    filterValue.addEventListener('input', filterFunction);
+
+    filterMenuContainer.append(
+      btnClean,
+      filterLabel,
+      filterOperator,
+      filterValue,
+      btnExit
+    );
+    this.tableHeader.append(filterMenuContainer);
+  }
+
+  createPagination(totalAmount = 0) {
+    const paginationContainer = document.createElement('div');
+    paginationContainer.classList.add('pagination-container');
+
+    const dropdownPagintaionContainer = document.createElement('select');
+    dropdownPagintaionContainer.classList.add('dropdown-pagination-container');
+
+    this.pageNumberContainer = document.createElement('ul');
+    this.pageNumberContainer.classList.add('page-num-container');
+
+    const dropdownPaginationItems = this.paginationConfigList.map((item) => {
+      const dropPagItem = document.createElement('option');
+      dropPagItem.classList.add('drop-pag-item');
+      dropPagItem.textContent = item.pageSize;
+
+      dropdownPagintaionContainer.addEventListener('change', (e) => {
+        if (this.amountElOnPage !== parseFloat(e.target.value)) {
+          this.amountElOnPage = parseFloat(e.target.value);
+          this.renderPagesAmount(totalAmount);
+          if (this.callbacksMap.paginationCallback) {
+            this.callbacksMap.paginationCallback();
+          }
+        }
+      });
+
+      return dropPagItem;
+    });
+
+    dropdownPagintaionContainer.append(...dropdownPaginationItems);
+    paginationContainer.append(
+      dropdownPagintaionContainer,
+      this.pageNumberContainer
+    );
+    this.tableContainer.append(paginationContainer);
+  }
+
+  renderPagesAmount(totalAmount = 0) {
+    const pages = Math.ceil(totalAmount / this.amountElOnPage);
+
+    for (let i = 1; i <= pages; i++) {
+      const pageNumberBtn = document.createElement('li');
+      pageNumberBtn.classList.add('page-number');
+      pageNumberBtn.textContent = i;
+      this.pageNumberContainer.append(pageNumberBtn);
+
+      pageNumberBtn.addEventListener('click', () => {
+        const currentPage = +pageNumberBtn.innerHTML;
+        if (this.callbacksMap.paginationCallback) {
```

Comment:
The same here
Before adding a listener, let's make sure, that we have a callback for this listener